homeassistant/components/elmax/__init__.py

```diff
@@ -104,6 +105,12 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         update_interval=timedelta(seconds=POLLING_SECONDS),
     )
 
+    async def _async_on_hass_stop(_: Event) -> None:
+        """Close connection when hass stops."""
+        await coordinator.async_shutdown()
+
+    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, _async_on_hass_stop)
```

Comment:
We need to cancel this listener once the entry is unloaded.

```suggestion
    entry.async_on_unload(
        hass.bus.async_listen_once(
            EVENT_HOMEASSISTANT_STOP, _async_on_hass_stop
        )
    )
```

Reply:
Right. Handled in 4e3c26fc0d.